packages/synapse-react-client/src/components/SynapseTable/SynapseTable.tsx

```diff
@@ -1,889 +1,275 @@
-        distinctEntityIds.add(rowSynapseId)
-      })
-    }
-    const distinctUserIds = getUniqueEntities(
-      data,
-      mapUserIdToHeader,
-      userIdColumnIndicies,
-    )
-    if (distinctEntityIds.size === 0 && distinctUserIds.size === 0) {
-      return
-    }
-    this.setState({
-      isFetchingEntityHeaders: true,
-    })
-    // Make call to resolve entity ids
-    if (distinctEntityIds.size > 0) {
-      const referenceList: ReferenceList = Array.from(distinctEntityIds).map(
-        id => {
-          return { targetId: id }
-        },
-      )
-      try {
-        const data = await SynapseClient.getEntityHeaders(
-          referenceList,
-          this.props.synapseContext.accessToken,
-        )
-        const { results } = data
-        results.forEach(el => {
-          mapEntityIdToHeader[el.id] = el
-        })
-      } catch (err) {
-        console.error('Error on retrieving entity header list , ', err)
-      }
-    }
-    const userProfileIds: string[] = []
-    if (distinctUserIds.size > 0) {
-      // Make call to get group headers and user profiles
-      const ids = Array.from(distinctUserIds)
-      // TODO: Grab Team Badge
-      try {
-        const data = await SynapseClient.getGroupHeadersBatch(
-          ids,
-          this.props.synapseContext.accessToken,
-        )
-        data.children.forEach(el => {
-          if (el.isIndividual) {
-            userProfileIds.push(el.ownerId)
-          } else {
-            mapUserIdToHeader[el.ownerId] = el
-          }
-        })
-      } catch (err) {
-        console.error('Error on getGroupHeaders batch: ', err)
-      }
-    }
-    if (userProfileIds.length > 0) {
-      try {
-        const data = await getUserProfileWithProfilePicAttached(userProfileIds)
-        data.list.forEach((el: UserProfile) => {
-          mapUserIdToHeader[el.ownerId] = el
+  const isShowingAccessColumn: boolean = Boolean(
+    showAccessColumn &&
+      entity &&
+      isEntityViewOrDataset(entity) &&
+      allRowsHaveId(data),
+  )
+
+  const rowsAreDownloadable =
+    entity && isFileViewOrDataset(entity) && isLoggedIn && allRowsHaveId(data)
+
+  const isShowingDirectDownloadColumn = Boolean(
+    rowsAreDownloadable && showDirectDownloadColumn,
+  )
+
+  const isShowingAddToV2DownloadListColumn: boolean = Boolean(
+    rowsAreDownloadable &&
+      !hideAddToDownloadListColumn &&
+      !isRowSelectionVisible,
+  )
+
+  const columns: ColumnDef<Row, string | null>[] = useMemo(
+    () => [
+      rowSelectionColumn,
+      addToDownloadListColumn,
+      directDownloadColumn,
+      accessColumn,
+      ...(data?.columnModels?.map((columnModel, index) => {
+        return columnHelper.accessor(row => row.values[index], {
+          id: columnModel.name,
+          enableSorting: isSortableColumn(columnModel.columnType),
+          enableResizing: true,
+          header: TableDataColumnHeader,
+          cell: TableDataCell,
```

Comment:
Components for column headers and cells are defined in SynapseTableRenderers.tsx